build-system/git.js

```diff
@@ -25,11 +25,20 @@ const {getStdout} = require('./exec');
  * Returns the branch point of the current branch off of master.
  * @return {string}
  */
-exports.gitBranchPoint = function() {
+exports.gitBranchPointFromMaster = function() {
+  return getStdout('git merge-base master HEAD^').trim();
```

Comment:
There's a nasty bug here. The branch point must be calculated based on `HEAD`, not `HEAD^`, or else the local changes logic breaks, and changes from other commits are included in the list of files changed on a given branch. I'm trying to dig into why this was changed to `HEAD^`.

Reply:
Found the culprit: #18940.

Reply:
Hmm... I can revert that PR, but I wonder if that's just gonna break a new thing. Git is confusing. Let's talk about this offline?

Reply:
Fixed with #19933.